File: objective-c/ArcGISSample/RotationRenderer/RotationRendererViewController.m
//
//  RotationRendererViewController.m
//  ArcGISSample
//
//  Created by esrij on 2015/07/27.
//  Copyright (c) 2015年 esrij. All rights reserved.
//

#import "RotationRendererViewController.h"

@implementation RotationRendererViewController


- (void)viewDidLoad
{
    [super viewDidLoad];
    
    AGSMapView *agsMapView = [[AGSMapView alloc] initWithFrame:self.view.bounds];
	[self.view addSubview:agsMapView];

    //タイルマップサービスレイヤーの追加
	NSURL *url = [NSURL URLWithString:@"http://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer"];
	AGSTiledMapServiceLayer *tiledMapServiceLayer = [[AGSTiledMapServiceLayer alloc] initWithURL:url];
    [agsMapView addMapLayer:tiledMapServiceLayer withName:@"Tiled Layer"];
    
    //シンボル表示用のフィーチャレイヤーの表示
    NSURL *flayerUrl = [NSURL URLWithString:@"http://tmservices1.esri.com/arcgis/rest/services/LiveFeeds/NOAA_METAR_current_wind_speed_direction/MapServer/0"];
    AGSFeatureLayer *flayer = [AGSFeatureLayer featureServiceLayerWithURL:flayerUrl mode:AGSFeatureLayerModeOnDemand];
    flayer.outFields = [NSArray arrayWithObjects: @"WIND_DIRECT", nil];
    [agsMapView addMapLayer:flayer withName:@"Feature Layer"];
    
    //画像ファイルを指定してシンボルとレンダラーを作成
    UIImage *image = [UIImage imageNamed:@"ArcGIS.bundle/LocationDisplayCourse@2x.png"];
    AGSPictureMarkerSymbol *pointSymbol = [[AGSPictureMarkerSymbol alloc] initWithImage:image];
    AGSSimpleRenderer *renderer = [[AGSSimpleRenderer alloc] initWithSymbol:pointSymbol];
    
    //レンダラーの回転角度の値をフィーチャレイヤーの"WIND_DIRECT"フィールドから取得
    renderer.rotationType = AGSRotationTypeGeographic;
    renderer.rotationExpression = @"[WIND_DIRECT]";
    
    //レンダラーをフィーチャレイヤーに適用
    flayer.renderer = renderer;

}


- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: swift/ArcGISSample/RotationRendererViewController.swift
//
//  RotationRendererViewController.swift
//  ArcGISSample
//
//  Created by esrij on 2015/07/27.
//  Copyright (c) 2015年 esrij. All rights reserved.
//

import UIKit
import ArcGIS


class RotationRendererViewController: UIViewController, AGSWebMapDelegate {
    
    override func viewDidLoad() {
        
        super.viewDidLoad()
        
        let agsMapView = AGSMapView(frame: view.bounds)
        view.addSubview(agsMapView)
        
        //タイルマップサービスレイヤーの追加
        let url = URL(string: "https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer")
        let tiledLyr = AGSTiledMapServiceLayer(url:url)
        agsMapView.addMapLayer(tiledLyr, withName:"Tiled Layer")
        
        //シンボル表示用のフィーチャレイヤーの表示
        let flayerUrl = URL(string: "https://tmservices1.esri.com/arcgis/rest/services/LiveFeeds/NOAA_METAR_current_wind_speed_direction/MapServer/0")
        let flayer = AGSFeatureLayer(url: flayerUrl, mode: .onDemand)
        flayer?.outFields = ["WIND_DIRECT"]
        agsMapView.addMapLayer(flayer, withName:"Feature Layer")

        //画像ファイルを指定してシンボルとレンダラーを作成
        let image = UIImage(named: "ArcGIS.bundle/LocationDisplayCourse@2x.png")
        let pointSymbol = AGSPictureMarkerSymbol(image: image)
        let renderer = AGSSimpleRenderer(symbol: pointSymbol)

        //レンダラーの回転角度の値をフィーチャレイヤーの"WIND_DIRECT"フィールドから取得
        renderer?.rotationType = .geographic
        renderer?.rotationExpression = "[WIND_DIRECT]"
        
        //レンダラーをフィーチャレイヤーに適用
        flayer?.renderer = renderer
        
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
}
